aspnetcore/security/cookie-sharing/samples/WebCookieShare-NetFx/App_Start/Startup.Auth.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IO;
+using Microsoft.AspNet.Identity;
+using Microsoft.AspNet.Identity.Owin;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Owin;
+using Microsoft.Owin.Security.Cookies;
+using Microsoft.Owin.Security.Google;
+using Microsoft.Owin.Security.Interop;
+using Owin;
+using WebCookieShare_NetFx.Models;
+
+namespace WebCookieShare_NetFx
+{
+    public partial class Startup
+    {
+        // For more information on configuring authentication, please visit https://go.microsoft.com/fwlink/?LinkId=301864
+        public void ConfigureAuth(IAppBuilder app)
+        {
+            // Configure the db context, user manager and signin manager to use a single instance per request
+            app.CreatePerOwinContext(ApplicationDbContext.Create);
+            app.CreatePerOwinContext<ApplicationUserManager>(ApplicationUserManager.Create);
+            app.CreatePerOwinContext<ApplicationSignInManager>(ApplicationSignInManager.Create);
+
+            // Enable the application to use a cookie to store information for the signed in user
+            // and to use a cookie to temporarily store information about a user logging in with a third party login provider
+            // Configure the sign in cookie
+            #region snippet_netfx_cookie_auth
+            app.UseCookieAuthentication(new CookieAuthenticationOptions
+            {
+                LoginPath = new PathString("/Account/Login"),
+                Provider = new CookieAuthenticationProvider
+                { 
+                    OnValidateIdentity = SecurityStampValidator.OnValidateIdentity<ApplicationUserManager, ApplicationUser>(
```

Comment:
This requires you to have a shared user database between AspNetCore Identity and AspNet Identity? @HaoK

Reply:
Yea, the shared database piece is something that I haven't tried to get working yet. Most likely, customers will need to sign in from *one* of the apps (though either should be fine) and then just use the cookie or log out from both apps after that.

Reply:
SecurityStampValidator wouldn't work for one of the apps in that case.

Reply:
In other samples like this we've done before (like in [System.Web adapter samples](https://github.com/dotnet/systemweb-adapters/blob/main/samples/MvcApp/App_Start/Startup.Auth.cs), we've just had users login using the ASP.NET app (so this works) and enabled the ASP.NET Core app to use the cookie or log the user out.

I'll update the docs to point out that users should only be signing in through one of the apps so that they don't have to worry about sharing a database.

Reply:
I added a couple sentences at the end of this section clarifying that users should still only *sign in* to one of the apps. But the configuration explained here will allow the cookie generated by that app to be *used* in either the ASP.NET or ASP.NET Core app. @HaoK, please let me know if you see anything in the new addition that you would change.

Reply:
Looks great to me!